### ISSUE TITLE
ParaViewWeb Visualizer installation with python3 ParaView binary

### ISSUE THREAD

**Issue by Shaileshbg1** [Kitware/visualizer#77]
Hello,

I built ParaView binaries on python3 but that does not have the python launcher nor the visualiser front end installed in it. Could you please let me know on how to install all the necessary packages to use ParaViewWeb visualizer with the python3 ParaView binaries.

Thanking you,

Regards,
Shailesh

**Comment by Shaileshbg1** [comment]
Hi Jourdain,

I am very sorry to disturb you again with my doubts. I built ParaView version 5.5.2 based on python 3.5, but the built binary does not have the file ***/visualizer/server/pvw-visualizer.py in the ParaView tree as well as ***/visualizer/www/index.html. There is no wslink installed itself.

Are there any flags which I have to switch on during cmake to enable the build of visualizer while building ParaView itself. Or is it possible to patch my version of ParaView so it can work with the ParaViewWeb Visualizer app. Or is there a way in which I can clone and build wslink and integrate it with my ParaView. 

Any suggestions would be greatly appreciated. Looking forward to hearing back from you.

Regards,
Shailesh

**Comment by jourdain** [comment]
When building ParaView, you can use the [superbuild](https://gitlab.kitware.com/paraview/paraview-superbuild) or the ParaView repo itself that you used. When using the ParaView repo, you will be missing all the dependencies. For Visualizer, you can use your current build by just creating a Python virtual environment and installing wslink in it. For Visualizer, you should get the bundle from npm (https://registry.npmjs.org/pvw-visualizer/-/pvw-visualizer-3.1.10.tgz).

Then you will run it using a command line like follow:

```
/.../pvpython ./pvw-visualizer-3.1.10/server/pvw-visualizer.py --content ./pvw-visualizer-3.1.10/dist --virtual-env /.../path/to/your/virtual-env/base/directory [...]
```

**Comment by Shaileshbg1** [comment]
Hi Jourdain,

Thank you for your reply. While building ParaView using superbuild where do I specify the python3 version. I went through the CMakeLists.txt but could not find any flag pertaining to this. 
Also how do I specify the other cmake variables which specify qt to be off. In the documentation it says 
"ParaView supports multiple rendering engines including `egl`, `mesa`, `osmesa`, and `qt5`. All of these are incompatible with each other. If none of these are chosen, a UI-less ParaView will be built (basically just `pvpython`).", so if I do not specify any rendering engine will it serve my purpose.
I found this in mailing list where someone had tried to set these variables, but it did not work for him.

  -DPYTHON_LIBRARIES=/opt/python361/lib/libpython3.6m.a \
  -DPYTHON_INCLUDE_DIRS=/opt/python361/include \

Looking forward to hearing back from you.

Regards,
Shailesh
